kobo-uncaged/kutypes.go

```diff
@@ -195,3 +156,43 @@ func (d koboDevice) FullCover() image.Point {
 		return image.Pt(600, 800)
 	}
 }
+
+type koboCover int
+
+const (
+	fullCover koboCover = iota
+	libFull
+	libGrid
+)
+
+func (k koboCover) String() string {
+	switch k {
+	case fullCover:
+		return "N3_FULL"
+	case libFull:
+		return "N3_LIBRARY_FULL"
+	case libGrid:
+		return "N3_LIBRARY_GRID"
+	default:
+		panic("unknown cover type")
```

Comment:
Just a heads up that I'm generally not a fan of panicing. Especially since we're being run from a rather delicate shell script, where its nice to have a known return value.

In this case though, I guess its fine. There should be no reason this ever occurs, except for if we make a typo in the code somewhere.

Reply:
That's the same way I am about it. I only panic if it is an issue with the function's own code. I never panic if it could be triggered by an external library, if it is a runtime error, or in the case of a library if it could be triggered by the user. The only exception I make is if it is a one-off command or if it is dangerous to continue running.

Reply:
That makes sense.